parse/ShipDesignsParser.cpp

```diff
@@ -120,26 +169,22 @@ namespace {
 }
 
 namespace parse {
-    bool ship_designs(const boost::filesystem::path& path, std::map<std::string, ShipDesign*>& designs)
-    { return detail::parse_file<rules, std::map<std::string, ShipDesign*>>(path, designs); }
-
-    bool ship_designs(std::map<std::string, ShipDesign*>& designs) {
+    bool ship_designs(const boost::filesystem::path& path, std::map<std::string, std::unique_ptr<ShipDesign>>& designs) {
         bool result = true;
-
-        for(const boost::filesystem::path& file : ListScripts("scripting/ship_designs")) {
-            result &= detail::parse_file<rules, std::map<std::string, ShipDesign*>>(file, designs);
+        for(const boost::filesystem::path& file : ListScripts(path)) {
+            try {
+                result &= detail::parse_file<rules, std::map<std::string, std::unique_ptr<ShipDesign>>>(file, designs);
+            } catch (const std::runtime_error& e) {
+                result = false;
+                ErrorLogger() << "Failed to parse ship design in " << file << " from " << path << " because " << e.what();;
+            }
         }
-
         return result;
     }
 
-    bool monster_designs(std::map<std::string, ShipDesign*>& designs) {
-        bool result = true;
-
-        for (const boost::filesystem::path& file : ListScripts("scripting/monster_designs")) {
-            result &= detail::parse_file<rules, std::map<std::string, ShipDesign*>>(file, designs);
-        }
+    bool ship_designs(std::map<std::string, std::unique_ptr<ShipDesign>>& designs)
+    { return ship_designs("scripting/ship_designs", designs); }
 
-        return result;
-    }
+    bool monster_designs(std::map<std::string, std::unique_ptr<ShipDesign>>& designs)
+    { return ship_designs("scripting/monster_designs", designs); }
```

Comment:
I misinterpreted the question.

Yes it is portable.

The POSIX forward slash in pathnames '/'  is referred to as the generic format and must be accepted by all conforming implementations of filesystem for all systems.